### PR TITLE
use HABTM for zones and shipping methods

### PR DESCRIPTION
**Pull request by BenMorganIO**

### CODE REVIEW

File: core/app/models/spree/shipping_method_zone.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module Spree
+  class ShippingMethodZone < Spree::Base
+    belongs_to :zone
```
Comment (jhawthorn) [inline core/app/models/spree/shipping_method_zone.rb]: Same question as other PR: Should we have an inverse of here, or is that detected successfully?

Comment (BenMorganIO) [inline core/app/models/spree/shipping_method_zone.rb]: I would assume it is...

**Comment by jhawthorn** [comment]
:+1: I checked and it is. LGTM

**Comment by magnusvk** [comment]
:+1: thanks for pulling this out

**Comment by BenMorganIO** [comment]
There's no dependent destroy added here. Also, what do you guys think of back tracking and adding foreign keys in this PR?

**Pull request merged**